src/app/notifications/notifications-list/notifications-list.component.ts

```diff
@@ -0,0 +1,36 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {NotificationId} from '../../entities/notification';
+import {AngularFirestore} from '@angular/fire/firestore';
+import {Observable} from 'rxjs';
+import {NotificationsService} from '../notifications.service';
+import {flatMap} from 'rxjs/operators';
+import {UserData} from '../../entities/user';
+
+@Component({
+  selector: 'app-notifications-list',
+  templateUrl: './notifications-list.component.html',
+  styleUrls: ['./notifications-list.component.scss']
+})
+export class NotificationsListComponent implements OnInit {
+  @Input() user: Observable<UserData>;
+  notifications$: Observable<NotificationId[]>;
+
+  constructor(private notifications: NotificationsService, private afs: AngularFirestore) { }
+
+  ngOnInit() {
+    this.notifications$ = this.user.pipe(
+      flatMap(user => this.notifications.getNotificationsForUser(user.uid))
+    );
+  }
+
+  seen(event: MouseEvent, notification: NotificationId) {
+    event.preventDefault();
+    this.updateNotificationState(notification.id, true);
+  }
+
+  updateNotificationState(notificationId: string, seen: boolean) {
```

Comment:
Why here and not in DbService?

Reply:
we did the same separation in Gamification module, in order to keep all related logical functions together and inside its module.
However, you are right and it should not be in the component ts file. I will move the function to NotificationService.